### ISSUE TITLE
Email bouncing back

### ISSUE THREAD

**Issue by Nhor** [mailcow/mailcow-dockerized#182]
Hello, I've set up an email at my machine, configured the DNS and I can successfully send an email, but I can't receive any, because every email I receive is bouncing back. Here's my MX setup:

![mx](https://cloud.githubusercontent.com/assets/11555841/24841199/83c5b680-1d7e-11e7-9bf3-105f1b6b7fc4.png)

And log from `mailcowdockerized_postfix-mailcow_1` docker container:
```
Apr  9 21:37:42 mail postfix/smtp[124]: warning: host mail.kwrzesinska.com[85.xxx.x.xxx]:25 greeted me with my own hostname mail.kwrzesinska.com
Apr  9 21:37:42 mail postfix/smtp[124]: warning: host mail.kwrzesinska.com[85.xxx.x.xxx]:25 replied to HELO/EHLO with my own hostname mail.kwrzesinska.com
Apr  9 21:37:42 mail postfix/smtp[124]: B418AA5291: to=<kontakt@kwrzesinska.com>, relay=mail.kwrzesinska.com[85.xxx.x.xxx]:25, delay=0.17, delays=0.14/0.03/0.01/0, dsn=5.4.6, status=bounced (mail for kwrzesinska.com loops back to myself)
```

I've noticed that the postfix `main.cf` file has some strange config:

```
myhostname = bbdeb8fab105.ec2.internal
mydestination = $myhostname, bbdeb8fab105, localhost.localdomain, , localhost
```

but I tried to change `myhostname` to `domain.com` and `mail.domain.com` with no success. I've ran out of ideas after trying to solve it the whole day, so I'm asking some broader community for help. Thanks!

**Comment by andryyy** [comment]
Hi, It is important to change myhostname back.
Are you trying to send mail from a machine with the same hostname?

**Comment by Nhor** [comment]
Hey, I am sending an email from a machine on which my mail server with iRedMail runs. It uses completely different machine and domain. I changed the hostname back to `bbdeb8fab105.ec2.internal` after I found out none of my alternatives worked.

**Comment by andryyy** [comment]
The machine "helos" your mailcow stack with its own hostname. 
I sent you a mail but it bounced: Diagnostic-Code: X-Postfix; mail for kwrzesinska.com loops back to myself

"mydestination" should not contain a domain which you setup in mailcow UI, that's where this error comes from. In case you removed it, you should restart Postifx with `docker-compose restart postfix-mailcow`.

**Comment by Nhor** [comment]
That's what I did, but maybe I overlooked something. I don't have access to my machine right now, so I will try once more in a couple of hours, thanks!

**Comment by andryyy** [comment]
You can also try to copy the original main.cf back if it still fails. :-)

**Comment by Nhor** [comment]
I copied `data/conf/postfix/main.cf`  to `mailcowdockerized_postfix-mailcow_1:/etc/postfix/main.cf` and restarted the container with `docker-compose restart postfix-mailcow` command.  Unfortunately incoming emails are still being bounced. Maybe I'm lacking some host machine config and the postfix container is fine?

**Comment by andryyy** [comment]
You don't need to copy it, the containers read them "live" from your filesystem.
Can you post the output of `docker-compose exec postfix-mailcow postconf`?

**Comment by Nhor** [comment]
[postconf.txt](https://github.com/andryyy/mailcow-dockerized/files/911135/postconf.txt)

**Comment by andryyy** [comment]
`docker-compose exec postfix-mailcow postconf -e mydomain=mail.kwrzesinska.com` should fix it.

Try not to mess with the files in /var/lib/docker/volumes/bla, just change the files in data/conf. :)

**Comment by Nhor** [comment]
This is very embarassing, but after trying all you proposed without success I removed everything and set up the environment once again. During domain addition in the mailbox administration I realized that last time I checked the "Backup MX" options. Thanks for your help and sorry for wasting your time, mailcow is a great piece of software which I chose in favor of previously used iRedMail because it stroke me to be so easy to set up with docker. And apparently it is!
